client/src/components/GiveForm.js

```diff
@@ -0,0 +1,78 @@
+import React, { useState } from "react";
+import { useForm } from "react-hook-form";
+import { Redirect } from "react-router-dom";
+import { ErrorMessage } from "@hookform/error-message";
+
+import WarmFuzzyClient from "../apiClient/WarmFuzzyClient";
+import Authentication from "../providers/Authentication";
+
+const GiveForm = () => {
+  const { user, createNewFuzzy } = Authentication.useAuthentication();
+  const [shouldRedirect, setShouldRedirect] = useState(false);
+  const { handleSubmit, register, errors, reset } = useForm();
+
+  const onSubmit = async (data) => {
+    const client = new WarmFuzzyClient();
+    const warmFuzzyData = await client.postFuzzy({ ...data, elevatorId: user.userId });
```

Comment:
we don't want to set this client side as it makes us subject to impersonation attacks: https://www.mimecast.com/content/impersonation-attack/

Basically, a malicious user could hack the request and pretend to be someone they are not. We should use the session context on the server side to definitively determine who the elevator is.

Reply:
I would like to talk through this pattern together.